src/avk.cpp

```diff
@@ -2314,11 +2314,17 @@ namespace avk
 		
 		return result;
 	}
-	
+
 	std::optional<command_buffer> buffer_t::fill(const void* aDataPtr, size_t aMetaDataIndex, sync aSyncHandler)
 	{
 		auto metaData = meta_at_index<buffer_meta>(aMetaDataIndex);
 		auto bufferSize = static_cast<vk::DeviceSize>(metaData.total_size());
+		return fill_partially(aDataPtr, bufferSize, std::move(aSyncHandler));
+	}
+
+	std::optional<command_buffer> buffer_t::fill_partially(const void* aDataPtr, size_t aDataSizeInBytes, sync aSyncHandler)

```

Comment:
This only allows to fill the beginning of a buffer. However, from a `fill_partially` method, I would expect to be able to fill arbitrary locations within this buffer. => An `size_t aOffset` parameter should be added and put to proper use.

Furthermore, it seems a bit odd that the meta-index is no longer required for  `fill_partially`. I mean, actually it is not really required. But maybe we should add it anyways for consistency reasons and because we could implement a check whether the given range to be filled is really within the bounds of this buffer (but only `#if _DEBUG`).

Reply:
After adding the offset and meta index params, the signatures of `fill` and `fill_partially` are different again, so if you prefer we can rename `fill_partially` as an overload to `fill`.